src/Microsoft.Dnx.Runtime/LockFileBasedProjectResolver.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+
+namespace Microsoft.Dnx.Runtime
+{
+    public class LockFileBasedProjectResolver : IProjectResolver
+    {
+        private readonly string _baseDirectory;
+        private readonly LockFile _lockFile;
+        private readonly IDictionary<string, ProjectInformation> _projects;
+
+        public LockFileBasedProjectResolver(LockFile lockFile, string projectDirectory, Project root)
+        {
+            _lockFile = lockFile;
+
+            _baseDirectory = projectDirectory;
+
+            _projects = GetSearchPaths(lockFile, projectDirectory);
+            _projects[root.Name] = new ProjectInformation(root);
+        }
+
+        public IEnumerable<string> SearchPaths
```

Comment:
Path to the `global.json` was not added to the `_searchPaths`. New? https://github.com/aspnet/dnx/blob/victorhu/require-servicing/src/Microsoft.Dnx.Runtime/ProjectResolver.cs#L71